src/prefect_server/api/agents.py

```diff
@@ -0,0 +1,142 @@
+
+    return await api.models.Agent(
+        tenant_id=tenant_id,
+        agent_config_id=agent_config_id,
+        labels=labels,
+        name=name,
+        type=type,
+        core_version=core_version,
+    ).insert()
+
+
+@register_api("agents.update_agent_last_queried")
+async def update_agent_last_queried(agent_id: str) -> bool:
+    """
+    Update an agent's last_queried value
+
+    Args:
+        - agent_id (str): the id of the agent to update
+
+    Returns:
+        - bool: whether the update was successful
+    """
+    if agent_id is None:
+        raise ValueError("Must supply an agent ID to update.")
+    result = await api.models.Agent.where(id=agent_id).update(
+        set={"last_queried": pendulum.now("utc")}
+    )
+    return bool(result.affected_rows)  # type: ignore
+
+
+@register_api("agents.delete_agent")
+async def delete_agent(agent_id: str) -> bool:
+    """
+    Delete an agent
+
+    Args:
+        - agent_id (str): the id of the agent to delete
+
+    Returns:
+        - bool: whether the delete was successful
+    """
+    if agent_id is None:
+        raise ValueError("Must supply an agent ID to delete.")
+    result = await api.models.Agent.where(id=agent_id).delete()
+    return bool(result.affected_rows)  # type: ignore
+
+
+@register_api("agents.create_agent_config")
+async def create_agent_config(
+    tenant_id: str,
+    name: str,
+    settings: dict,
+) -> str:
+    """
+    Creates an agent config, returning its id
+
+    Args:
+        - tenant_id (str): the tenant id
+        - name(str): the agent config name
+        - settings (dict): agent config settings
+
+    Returns:
+        - str: the agent config id
+    """
+    return await api.models.AgentConfig(
+        tenant_id=tenant_id, name=name, settings=settings
+    ).insert()
+
+
+@register_api("agents.delete_agent_config")
+async def delete_agent_config(agent_config_id: str) -> bool:
+    """
+    Delete an agent config
+
+    Args:
+        - agent_config_id (str): the id of the agent config to delete
+
+    Returns:
+        - bool: whether the delete was successful
+    """
+    if agent_config_id is None:
+        raise ValueError("Must supply an agent ID to delete.")
+    result = await api.models.AgentConfig.where(id=agent_config_id).delete()
+    return bool(result.affected_rows)  # type: ignore
+
+
+@register_api("agents.update_agent_config")
+async def update_agent_config(agent_config_id: str, name: str, settings: dict) -> str:
+    """
+    Update an agent config
+
+    Args:
+        - agent_config_id (str): the agent config ID
+        - name (str): the agent config name
+        - settings (dict): the settings to set on the agent config
+
+    Returns:
+        - bool: whether the update was successful
+    """
+    if not agent_config_id:
+        raise ValueError("Invalid agent config ID.")
+
+    update = {}
+    if name:
+        update["name"] = name
+    if settings:
+        update["settings"] = settings
```

Comment:
This sort of logic actually prevents users from being able to, e.g., clear their settings (because an empty dict or `None` is never respected as the input).  We could do `if settings is not None:` instead, and "clearing" settings simply means providing an empty dict to `settings`